packages/backend/src/core/DiscoveryWatcher.ts

```diff
@@ -69,19 +81,18 @@ export class DiscoveryWatcher {
     this.logger.info('Update finished', { blockNumber })
   }
 
-  async compareWithCommitted(
+  async findChanges(
     name: string,
-    discovered: AnalyzedData[],
+    discovery: ProjectParameters,
     overrides?: Record<string, DiscoveryContract>,
   ): Promise<DiscoveryDiff[]> {
-    const committed = await this.configReader.readDiscovery(name)
-    const parsedDiscovery = parseDiscoveryOutput(discovered)
+    const databaseEntry = await this.repository.findLatest(name)
 
-    return diffDiscovery(
-      committed.contracts,
-      parsedDiscovery.contracts,
-      overrides ?? {},
-    )
+    const currentContracts = databaseEntry
+      ? databaseEntry.discovery.contracts
+      : (await this.configReader.readDiscovery(name)).contracts
```

Comment:
In another PR: We should add config hash to discovery output so that if config changes we always prioritize discovery from disk.